### PR TITLE
Hcl2 rewrite

### PR DESCRIPTION
**Pull request by Stromweld**
## Description
Future proofing by rewriting code in HCL2 format and redoing github actions to lint and validate templates

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I have run the pre-merge tests locally and they pass.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by Stromweld** [comment]
vagrant box build tests and deployment coming in next PR

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,103 +1,199 @@
...
+            echo "::set-output name=files_changed::false"
+          fi
+      - name: markdown-link-check
+        if: steps.verify-changed-files.outputs.files_changed == 'true'
+        uses: gaurav-nelson/github-action-markdown-link-check@master
+        with:
+          use-verbose-mode: "yes"
+
+  yamllint:
+    uses: Stromweld/github-workflows/.github/workflows/yamllint.yml@main
```
Comment (tas50) [inline .github/workflows/ci.yml:33]: Was this source intentional?

File: .github/workflows/ci.yml.bak [hunk 2]
```diff
@@ -0,0 +1,103 @@
+---
+
+name: Packer
```
Comment (tas50) [inline .github/workflows/ci.yml.bak:3]: This file should probably just get deleted if it's not used anymore

File: NOTICE.md [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
 
 Chef Bento
-Copyright 2012-2019, Chef Software, Inc.
+Copyright 2012-2022, Progress Software Corporation
 
 The baseboxes in the "definitions" directory is from Tim Dysinger's
 "basebox" project.
 
-* https://github.com/dysinger/basebox
+* <https://github.com/dysinger>
```
Comment (tas50) [inline NOTICE.md:9]: If the link 404s it should just go away

File: NOTICE.md [hunk 4]
```diff
@@ -1,12 +1,12 @@
 # Bento NOTICE
 
 Chef Bento
-Copyright 2012-2019, Chef Software, Inc.
+Copyright 2012-2022, Progress Software Corporation
```
Comment (tas50) [inline NOTICE.md:4]: pretty sure you need both copyrights on this

**Comment by tas50** [comment]
@Stromweld Is there a plan to handle the creation of the box descriptions and the uploading of the slug boxes? That was all handled in the ruby app before.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Stromweld) [reply .github/workflows/ci.yml:33]: yes for now till I can get with the pipeline guys and find an official chef home them.

File: .github/workflows/ci.yml.bak (hunk 2 above)
Comment (Stromweld) [reply .github/workflows/ci.yml.bak:3]: Thanks for catching that

File: NOTICE.md (hunk 3 above)
Comment (Stromweld) [reply NOTICE.md:9]: ok

File: NOTICE.md (hunk 4 above)
Comment (Stromweld) [reply NOTICE.md:4]: thanks.

**Comment by Stromweld** [comment]
@tas50 Yes I left a comment above but my next PR will be focused on adding builds for integration tests and potentially saving them as artifacts in github actions then a CD workflow that will publish them to vagrant cloud. I was also looking into the packef-vagrant-cloud plugin that can publish boxes to a vagrant cloud box and version it, but it won't crate the cloud box definition for a new box just add the box under specific provider for an already created vagrant-cloud box.

**Comment by Stromweld** [comment]
I've got working pipelines that also test the builds for x86_64 in virtualbox and arm64 in parallels. Most builds are working for x86 a few more less used OS types still need some work/testing. Most arm64 builds needs work/testing. Would like to get this merged to main for visibility and hopefully get more help from the community on fixing the remainder builds.

**Comment by Stromweld** [comment]
To run a build simply run this command from in the bento repo folder and change the values for the provider and OS variables file for the machine you'd like to build. `packer build -only=virtualbox-iso.vm -var-file=os_pkrvars/almalinux/almalinux-9-x86_64.pkrvars.hcl ./packer_templates`

**Comment by Stromweld** [comment]
Follow on work will be around getting the rest of the builds working and adding more providers as well as CD publishing pipeline.

**Review by ashiqueps — APPROVED** [review]

**Pull request merged**